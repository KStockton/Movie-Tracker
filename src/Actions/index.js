export const addTopMovies = topMovies => ({
    type: "ADD_TOP_MOVIES",
    topMovies
})