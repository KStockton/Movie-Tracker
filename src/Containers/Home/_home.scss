.home-container {
  background: #000000; /* fallback for old browsers */
  background: -webkit-linear-gradient(
    to top,
    #434343,
    rgb(31, 31, 31)
  ); /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(
    to top,
    #0f0f0f,
    rgb(24, 24, 24)
  ); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
  overflow: hidden;
  height: 100%;
  color: white;
  overflow: scroll;
  height: 90%;
}

.home-title {
  padding: 1em 0;
}
