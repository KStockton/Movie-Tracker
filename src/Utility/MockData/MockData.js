
export const uncleanMovies =
[
  {
      "vote_count": 2045,
      "id": 19404,
      "video": false,
      "vote_average": 9,
      "title": "Dilwale Dulhania Le Jayenge",
      "poster_path": "/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg",
      "original_language": "hi",
      "original_title": "दिलवाले दुल्हनिया ले जायेंगे",
      "genre_ids": [
          35,
          18,
          10749
      ],
      "backdrop_path": "/nl79FQ8xWZkhL3rDr1v2RFFR6J0.jpg",
      "adult": false,
      "overview": "Raj is a rich, carefree, happy-go-lucky second generation NRI. Simran is the daughter of Chaudhary Baldev Singh, who in spite of being an NRI is very strict about adherence to Indian values. Simran has left for India to be married to her childhood fiancé. Raj leaves for India with a mission at his hands, to claim his lady love under the noses of her whole family. Thus begins a saga.",
      "release_date": "1995-10-20"
  },
  {
      "vote_count": 12962,
      "id": 278,
      "video": false,
      "vote_average": 8.7,
      "title": "The Shawshank Redemption",
      "poster_path": "/9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg",
      "original_language": "en",
      "original_title": "The Shawshank Redemption",
      "genre_ids": [
          18,
          80
      ],
      "backdrop_path": "/j9XKiZrVeViAixVRzCta7h1VU9W.jpg",
      "adult": false,
      "overview": "Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison, where he puts his accounting skills to work for an amoral warden. During his long stretch in prison, Dufresne comes to be admired by the other inmates -- including an older prisoner named Red -- for his integrity and unquenchable sense of hope.",
      "release_date": "1994-09-23"
  }]

  export const cleanMovies = 
  [{
     id: 19404,
     title: "Dilwale Dulhania Le Jayenge",
     backdrop_path: "/nl79FQ8xWZkhL3rDr1v2RFFR6J0.jpg",
     overview: "Raj is a rich, carefree, happy-go-lucky second generation NRI. Simran is the daughter of Chaudhary Baldev Singh, who in spite of being an NRI is very strict about adherence to Indian values. Simran has left for India to be married to her childhood fiancé. Raj leaves for India with a mission at his hands, to claim his lady love under the noses of her whole family. Thus begins a saga.",
     poster_path: "/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg",
     vote_average: 9,
     release_date: "1995-10-20",
     favorited: false
  },
    {
      id: 278,
      title:  "The Shawshank Redemption",
      backdrop_path: "/j9XKiZrVeViAixVRzCta7h1VU9W.jpg",
      overview: "Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison, where he puts his accounting skills to work for an amoral warden. During his long stretch in prison, Dufresne comes to be admired by the other inmates -- including an older prisoner named Red -- for his integrity and unquenchable sense of hope.",
      poster_path: "/9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg",
      vote_average: 8.7,
      release_date: "1994-09-23",
      favorited: false
  }]

  export const movieProp = {
    state: {
      movie: {
        id: 2341,
        title: "HellBoy",
        backdrop_path: "/5BkSkNtfrnT",
        favorited: false,
        overview: "walk on earth",
        poster_path: "/bk8LyaMqUtaQ9hUShuvFznQYQKR.jpg",
        release_date: "2019-04-12"
      }
    }
  }
