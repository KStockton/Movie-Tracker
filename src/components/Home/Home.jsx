import React from 'react'

const  Home = ()=>  {
  return (
    <section className="home-container">
        <h1>home</h1>
      
    </section>
  )
}

export default Home