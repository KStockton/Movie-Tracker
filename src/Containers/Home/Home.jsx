import React, { Component } from "react";
import { APIkey } from "../../Utility/Config/Key";
import { getTopMovies } from "../../Utility/Helpers/GetTopMovies";
import { connect } from "react-redux";
import { addTopMovies } from "../../Actions/index";
import CardContainer from "../../Components/CardContainer/CardContainer";

export class Home extends Component {
  constructor() {
    super();
    this.state = {
      topMovies: [],
      moviesArray: []
    };
  }
  async componentDidMount() {
    let movies = await getTopMovies(APIkey);
    this.props.addTopMovies(movies);
  }

  render() {
    return (
      <section className="home-container">

        <h2 className="home-title"> </h2>

        { this.props.topMovies.length > 0 ? (
          <CardContainer
            topMovies={this.props.topMovies}
            user={this.props.user}
          />
        ) : (
          <div />
        )}
      </section>
    );
  }
}
export const mapStateToProps = state => ({
  topMovies: state.topMovies,
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  addTopMovies: movies => dispatch(addTopMovies(movies))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);
