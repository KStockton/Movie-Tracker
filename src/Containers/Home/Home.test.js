import React from 'react';
import { Home, mapStateToProps, mapDispatchToProps } from './Home';
import { shallow } from 'enzyme';
import { addTopMovies } from '../../Actions/index';
import * as MD from '../../Utility/MockData/MockData'

const mockTopMovies = jest.fn()

describe('Home', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow (<Home addTopMovies={mockTopMovies} topMovies={MD.cleanMovies} user={MD.mockUser}/>)
  })

  it('Home component should match the snapshot', () =>{
    expect(wrapper).toMatchSnapshot()
  })
  it('should have a default state', () => {
    expect(wrapper.state('topMovies')).toEqual([])
    expect(wrapper.state('moviesArray')).toEqual([])
  })

  it('should call getTopMovies when componentDidMount is called', async () => {
   await wrapper.instance().componentDidMount();
    expect(mockTopMovies).toHaveBeenCalled();
  })

  it('should call dispatch when using addTopMovies props function', () => {
    const mockDispatch = jest.fn()
    const actionToDispatch = addTopMovies(MD.cleanMovies)

    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.addTopMovies(MD.cleanMovies)
    
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it('should return a props object with a user', () => {
    const mockState = {
      user: {
        id: 55,
        name: "nim",
        password: 'nimisawesome',
        email: "nim@sum.com"
      }
    }

    const expected = {
      user: mockState.user
    }

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  })

  it('should return a props object with a movies array', () => {
    const mockState = { topMovies: [{ 'title': "Space Jam" }]}
    const expected = {
      topMovies: mockState.topMovies
    }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  })

})     