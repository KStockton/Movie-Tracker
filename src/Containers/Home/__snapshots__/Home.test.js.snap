// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home component should match the snapshot 1`] = `
<section
  className="home-container"
>
  <h2
    className="home-title"
  >
     
  </h2>
  <CardContainer
    topMovies={
      Array [
        Object {
          "backdrop_path": "/nl79FQ8xWZkhL3rDr1v2RFFR6J0.jpg",
          "favorited": false,
          "id": 19404,
          "overview": "Raj is a rich, carefree, happy-go-lucky second generation NRI. Simran is the daughter of Chaudhary Baldev Singh, who in spite of being an NRI is very strict about adherence to Indian values. Simran has left for India to be married to her childhood fiancÃ©. Raj leaves for India with a mission at his hands, to claim his lady love under the noses of her whole family. Thus begins a saga.",
          "poster_path": "/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg",
          "release_date": "1995-10-20",
          "title": "Dilwale Dulhania Le Jayenge",
          "vote_average": 9,
        },
        Object {
          "backdrop_path": "/j9XKiZrVeViAixVRzCta7h1VU9W.jpg",
          "favorited": false,
          "id": 278,
          "overview": "Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison, where he puts his accounting skills to work for an amoral warden. During his long stretch in prison, Dufresne comes to be admired by the other inmates -- including an older prisoner named Red -- for his integrity and unquenchable sense of hope.",
          "poster_path": "/9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg",
          "release_date": "1994-09-23",
          "title": "The Shawshank Redemption",
          "vote_average": 8.7,
        },
      ]
    }
    user={
      Object {
        "email": "weOn@gmail",
        "favorites": Object {
          "state": Object {
            "movie": Object {
              "backdrop_path": "/5BkSkNtfrnT",
              "favorited": false,
              "id": 2341,
              "overview": "walk on earth",
              "poster_path": "/bk8LyaMqUtaQ9hUShuvFznQYQKR.jpg",
              "release_date": "2019-04-12",
              "title": "HellBoy",
            },
          },
        },
        "id": 3,
        "name": "Will",
        "password": "nimawesome",
      }
    }
  />
</section>
`;
